Trove.js

```diff
@@ -0,0 +1,337 @@
+			ZU.HTTP.doGet(textURL, function(text) {
+				item.notes.push({note: text.trim()});
+				item.complete();
+			});
+
+		});
+	});
+	translator.translate();	
+}
+
+var troveTypes = {
+	"Book": "book",
+	"Article Article/Book chapter": "bookSection",
+	"Thesis": "thesis",
+	"Archived website": "webpage",
+	"Conference Proceedings": "book",
+	"Audio book": "book",
+	"Article": "journalArticle",
+	"Article Article/Journal or magazine article": "journalArticle",
+	"Article Article/Conference paper": "conferencePaper",
+	"Article Article/Report": "report",
+	"Photograph": "artwork",
+	"Poster, chart, other": "artwork",
+	"Art work": "artwork",
+	"Object": "artwork",
+	"Microform Photograph": "artwork",
+	"Microform Object": "artwork",
+	"Sound": "audioRecording",
+	"Video": "videoRecording",
+	"Printed music": "book",
+	"Map": "map",
+	"Unpublished": "manuscript",
+	"Published": "document"
+}
+	
+
+function checkType(string) {
+	types = string.split("; ");
+	newString = types.join(" ");
+	if (newString in troveTypes) {
+		return troveTypes[newString];
+	} else {
+		while (types.length > 0) {
+			types.pop();
+			newString = types.join(" ");
+			if (newString in troveTypes) {
+				return troveTypes[newString];
+			}
+
+		}
+	}
+	return "book";
+}
+
+function cleanCreators(creators) {
+	newCreators = [];
+	for (var i = 0; i < creators.length; i++) {
+		var creator = creators[i];
+		creator.firstName = creator.firstName.replace(/\(?\d{4}-\d{0,4}\)?,?/, "").trim()
+		newCreators.push(creator);
+	}
+	return newCreators
+}
+
+function scrapeWork(doc, url) {
+	var thumbnailURL;
+
+	// Remove all params from url
+	var workURL = url.slice(0, url.indexOf('?'));
+	var bibtexURL = workURL + '?citationFormat=BibTeX';
+
+	// Need to get version identifier for the BibText url
+	var versionID = doc.body.innerHTML.match(/displayCiteDialog\(\'(.+?)\'/);
+	if (versionID !== null) {
+		bibtexURL += '&selectedversion=' + versionID[1];
+		thumbnailURL = ZU.xpathText(doc, "//a/img[@class='mosaic ui-shdw']/@src");
+	} else {
+		// It's a work -- so thumbnails are different
+		thumbnailURL = ZU.xpathText(doc, "//li[@class='imgfirst']//img/@src");
+	}
+
+	// Get the BibTex and feed it to the translator.
+	ZU.HTTP.doGet(bibtexURL, function(bibtex) {
+		var translator = Zotero.loadTranslator("import");
+		translator.setTranslator("9cb70025-a888-4a29-a210-93ec52da40d4");
+		translator.setString(bibtex);
+		translator.setHandler("itemDone", function (obj, item) {
+			item.itemType = checkType(item.type);
+			item.creators = cleanCreators(item.creators);
+
+			// This gives a better version-aware url.
+			item.url = ZU.xpathText(doc, "//meta[@property='og:url']/@content");
+			item.abstractNote = ZU.xpathText(doc, "//meta[@property='og:description']/@content");
+			delete item.itemID;
+			delete item.type;
+			if (thumbnailURL !== null) {
+				item.attachments.push({url: thumbnailURL, title: 'Trove thumbnail image', mimeType:'image/jpeg'});
```

Comment:
Trove is an aggregator -- images come from hundreds of different sources. Thumbnail links are the only reliable ones. I believe users will find them valuable as a quick check on the identity of the image. But I will also attach a link to the providing organisations where they are provided (this will be a link to an HTML page not to an image).